### PR TITLE
Fixed cmake warnings and set appropriate policy

### PR DESCRIPTION
**Pull request by aashish24**
Ref: #1011

### CODE REVIEW

**Comment by aashish24** [comment]
@doutriaux1  @sankhesh please review.

File: CMake/cmake_modules/ExternalProject.cmake [hunk 1]
```diff
@@ -156,6 +156,10 @@
 # (To distribute this file outside of CMake, substitute the full
 #  License text for the above reference.)
 
+# CMake policy usages
+cmake_policy(PUSH)
+cmake_policy(SET CMP0054 NEW)
```
Comment (jbeezley) [inline CMake/cmake_modules/ExternalProject.cmake]: This policy was introduced in 3.1.  Probably should add `if(POLICY CMP0054)` to support older cmake's.

Comment (aashish24) [inline CMake/cmake_modules/ExternalProject.cmake]: Sure @jbeezley. This is a good point. I pushed a change.

**Comment by doutriaux1** [comment]
error not related to this branch, messages are gone, merging in.

**Pull request merged**